system/lib/dlmalloc.c

```diff
@@ -850,6 +850,37 @@ extern "C" {
     /* ------------------- Declarations of public routines ------------------- */
     
 #ifndef USE_DL_PREFIX
+// XXX Emscripten XXX
+#if defined(__EMSCRIPTEN__) && !defined(__asmjs__)
+// XXX fastcomp HACK: in fastcomp, weak aliases loses to JavaScript versions of the function.
+// Therefore, we must define real functions. Remove this hack when fastcomp is removed.
```

Comment:
It's not super clear to me which code this comment says is a "hack". I think it's actually the old code down below in the `#else`? It would be good to clarify that.